dev/update_packages.dart

```diff
@@ -6,27 +6,39 @@
 import 'dart:io';
 
 final String binaryName = Platform.isWindows ? 'pub.bat' : 'pub';
-void update(Directory directory, bool upgrade) {
+int runPub(Directory directory, List<String> pubArgs) {
+  int updateCount = 0;
   for (FileSystemEntity dir in directory.listSync()) {
     if (dir is Directory) {
-      print("Updating ${dir.path}...");
+      updateCount++;
+      Stopwatch timer = new Stopwatch()..start();
+      stdout.write("Updating ${dir.path}...");
       ProcessResult result = Process.runSync(
         binaryName,
-        [ upgrade ? 'upgrade' : 'get' ],
+        pubArgs,
         workingDirectory: dir.path
       );
+      timer.stop();
+      stdout.write(" (${timer.elapsedMilliseconds} ms)");
       if (result.exitCode != 0) {
         print("... failed with exit code ${result.exitCode}.");
         print(result.stdout);
         print(result.stderr);
+      } else {
+        print(""); // Newline.
       }
     }
   }
+  return updateCount;
 }
 
 void main(List<String> arguments) {
+  Stopwatch timer = new Stopwatch()..start();
   bool upgrade = arguments.length > 0 && arguments[0] == '--upgrade';
   String FLUTTER_ROOT = new File(Platform.script.toFilePath()).parent.parent.path;
-  update(new Directory("$FLUTTER_ROOT/packages"), upgrade);
-  update(new Directory("$FLUTTER_ROOT/examples"), upgrade);
+  List<String> pubArgs = [ upgrade ? 'upgrade' : 'get' ];
+  int count = runPub(new Directory("$FLUTTER_ROOT/packages"), pubArgs);
```

Comment:
this is a total nit, but for code prettiness I'd have `int count = 0;` followed by two parallel `+=` lines. :-)